### PR TITLE
Fix SDP failure if hardware acceleration is not turned on

### PR DESCRIPTION
**Pull request by taste1981**

### CODE REVIEW

File: talk/owt/sdk/base/peerconnectiondependencyfactory.cc [hunk 1]
```diff
@@ -126,6 +128,13 @@ void PeerConnectionDependencyFactory::
     if (!GlobalConfiguration::GetCustomizedVideoDecoderEnabled()) {
       decoder_factory.reset(new MSDKVideoDecoderFactory());
     }
+  } else {
+    if (!encoded_frame_) {
```
Comment (jianjunz) [inline talk/owt/sdk/base/peerconnectiondependencyfactory.cc]: It's paired with line 144?

File: talk/owt/sdk/base/peerconnectiondependencyfactory.cc [hunk 2]
```diff
@@ -126,6 +128,13 @@ void PeerConnectionDependencyFactory::
...
+      encoder_factory = std::move(webrtc::CreateBuiltinVideoEncoderFactory());
+    }
+    if (!GlobalConfiguration::GetCustomizedVideoDecoderEnabled()) {
```
Comment (jianjunz) [inline talk/owt/sdk/base/peerconnectiondependencyfactory.cc]: Move to line 142?

File: talk/owt/sdk/base/peerconnectiondependencyfactory.cc (hunk 1 above)
Comment (taste1981) [reply talk/owt/sdk/base/peerconnectiondependencyfactory.cc]: paired with line 124.

File: talk/owt/sdk/base/peerconnectiondependencyfactory.cc (hunk 2 above)
Comment (taste1981) [inline talk/owt/sdk/base/peerconnectiondependencyfactory.cc]: this adds handling when hardware_acceleration is not true. not related with line 142.

Comment (taste1981) [reply talk/owt/sdk/base/peerconnectiondependencyfactory.cc]: refactored the code to be more reasonable here.

Comment (taste1981) [reply talk/owt/sdk/base/peerconnectiondependencyfactory.cc]: Refactored the code here to be more reasonable.

**Pull request merged**